addons/purchase/models/res_config_settings.py

```diff
@@ -58,3 +59,7 @@ def set_values(self):
             self.po_lock = po_lock
         if self.po_double_validation != po_double_validation:
             self.po_double_validation = po_double_validation
+
+    @api.depends('po_order_approval')
+    def _compute_po_approve_filter_visibility(self):
+        self.group_po_approve_filter_visibility = self.po_order_approval
```

Comment:
`po_order_approval` is defined on company_id, so `group_po_approve_filter_visibility` will take the value of the last company update on `po_order_approval`